capa/rules/__init__.py

```diff
@@ -1401,154 +1423,320 @@ def __getitem__(self, rulename):
     def __contains__(self, rulename):
         return rulename in self.rules
 
+    # unstable
     @staticmethod
-    def _index_rules_by_feature(rules) -> Tuple[Dict[Feature, Set[str]], List[str]]:
+    def _score_feature(scores_by_rule: Dict[str, int], node: capa.features.common.Feature) -> int:
+        """
+        Score the given feature by how "uncommon" it is, where a higher score is more uncommon.
+        Features that are not good for indexing will have a low score, or 0.
         """
-        split the given rules into two structures:
-          - "easy rules" are indexed by feature,
-            such that you can quickly find the rules that contain a given feature.
-          - "hard rules" are those that contain substring/regex/bytes features or match statements.
-            these continue to be ordered topologically.
 
-        a rule evaluator can use the "easy rule" index to restrict the
-        candidate rules that might match a given set of features.
+        #
+        # Today, these scores are manually assigned by intuition/experience/guesswork.
```

Comment:
can you include the range you envision here? 0-10 or similar

Reply:
resolved in [457cfe0](https://github.com/mandiant/capa/pull/2080/commits/457cfe018053f224af9ad6681e767aee1268fa6c)